stdlib/public/TensorFlow/Dataset.swift

```diff
@@ -139,3 +192,197 @@ extension SingleValueDatasetIterator : IteratorProtocol {
                                 output_shapes: [elementShape]))
   }
 }
+
+//===----------------------------------------------------------------------===//
+// Double value dataset
```

Comment:
Is this basically a copy-and-paste of the single value dataset, with some types changed?

Add a comment above SingleValueDataset and a comment above DoubleValueDataset describing this relationship and reminding people to keep them in sync. Also explain why they're duplicates, like you do in the PR description.